site/content/user-guide/amqp-adapter.md

```diff
@@ -3,28 +3,33 @@ title = "AMQP Adapter"
+* A failure caused by a client-side error (e.g invalid message address, content-type, adapter disabled for tenant, etc).
+* A failure caused by a server-side error (e.g sender queue full, no credit available yet to send the message, etc).
 
-In the case of a server-side error, the adapter responds with a *RELEASED* disposition, which indicates to the client that the message was OK but it cannot be delivered due to a failure beyond the control of the client. In this case, the client should attempt to redeliver the message again.
+For a client-side error, the adapter **rejects** the message and provides a reason why it was rejected. In the case of a server-side error, the adapter **releases** the message, indicating to the client that the message was OK but it cannot be delivered due to a failure beyond the control of the client. In this case, the client should attempt to redeliver the message again.
 
 <!--more-->
 
 ## Device Authentication
 
-By default, all Hono protocol adapters require clients (devices or gateway components) to authenticate during connection establishment. This is the preferred way for devices to publish data via protocol adapters. The AMQP adapter can be configured to *disallow* authentication for devices by setting the value of `HONO_AMQP_AUTHENTICATION_REQUIRED` to `false`. In this guide, we will give examples for publishing telemetry and events for *authenticated* and *unauthenticated* clients. We will see in the examples later that the main difference between publishing messages for authenticated and unauthenticated devices is in the format of the AMQP 1.0 **address**. For unauthenticated devices, the address should include the tenant and device identifier.
+By default, all Hono protocol adapters require clients (devices or gateway components) to authenticate during connection establishment. This is the preferred way for devices to publish data via protocol adapters. The AMQP adapter can be configured to *disallow* authentication for devices by setting the value of `HONO_AMQP_AUTHENTICATION_REQUIRED` to `false`. In this guide, we will give examples for publishing telemetry and events for *authenticated* and *unauthenticated* clients. We will see in the examples later that the main difference between publishing messages for authenticated and unauthenticated devices is in the format of the  ** message address**. For unauthenticated devices, the message address should include the tenant and device identifiers.
+
+For the purpose of demonstrating the usage of the AMQP adapter, the example deployment is shipped with a **minimal** command-line client to interact with the adapter. The client is package as a single executable JAR file and can be run directly on the command-line with *java -jar <jar-file>*. In the examples below, we will use this client to simulate a client device publishing messages to the adapter.
```

Comment:
`... client is packaged ...`

where can the client be found/downloaded?

Reply:
thanks.. I mentioned that it will be shipped with the example deployment. can we add the client to the example deployment? or should it be separate from Hono? the only new library added is the apache `commons-cli` package for defining and parsing command-line arguments

Reply:
When we want to add new dependencies to third party libraries, we need to get approval from the Eclipse IP team first. Getting this for Apache Commons libs is usually no big deal. However, I haven't seen any of the code so far. Did you create a PR for the client yet?

IMHO it should go into the `cli` module where we already have put the example consumer as well.

Reply:
No i did not create a PR yet. It is located on my github repo [here](https://github.com/Alfusainey/amqp-cli-client).

> IMHO it should go into the cli module where we already have put the example consumer as well.

Ok, i will put it there and create a PR. thanks!